base/collect_http.go

```diff
@@ -247,7 +250,9 @@ func (t *collectHTTPTask) run(exp *Experiment) error {
 			Description: "number of requests sent",
 			Type:        CounterMetricType,
 		}
-		in.updateMetric(m, mm, 0, float64(data.DurationHistogram.Count))
+		if err = in.updateMetric(m, mm, 0, float64(data.DurationHistogram.Count)); err != nil {
```

Comment:
I guess my question is: do we want to fail immediately if we can't update one metric of the set?  Or do we wish to continue in a best effort way?  Applies to all `updateMetric` calls

Reply:
This is a good point. For now, let us fail immediately (IMO, failing immediately, rather than ignoring silently is the improvement here). We can get a bit more experience with custom metrics, and then revisit this issue.

Reply:
Note that `collect_http` and `collect_grpc` have predefined metric names, which means update metric really should not fail, which means (if for some unknown reason it does), we really want to catch and return the error (so yes, failing immediately makes sense in these files).